oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/StorageID.java

```diff
@@ -0,0 +1,190 @@
+        }
+        fragments.add(new Fragment(new String[] {name}, String.class, false, value));
+        return this;
+    }
+
+    public StorageID append(String name, long value) {
+        if (StringUtil.isBlank(name)) {
+            throw new IllegalArgumentException("The name of storage ID should not be null or empty.");
+        }
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, name=" + name);
+        }
+        fragments.add(new Fragment(new String[] {name}, Long.class, false, value));
+        return this;
+    }
+
+    public StorageID append(String name, int value) {
+        if (StringUtil.isBlank(name)) {
+            throw new IllegalArgumentException("The name of storage ID should not be null or empty.");
+        }
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, name=" + name);
+        }
+        fragments.add(new Fragment(new String[] {name}, Integer.class, false, value));
+        return this;
+    }
+
+    public StorageID appendMutant(String[] source, long value) {
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, source=" + source);
+        }
+        fragments.add(new Fragment(source, Long.class, true, value));
+        return this;
+    }
+
+    public StorageID appendMutant(final String[] source, final String value) {
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, source=" + source);
+        }
+        fragments.add(new Fragment(source, String.class, true, value));
+        return this;
+    }
+
+    /**
+     * @return the string ID concatenating the values of {@link #fragments} by the underline(_).
+     */
+    public String build() {
+        sealed = true;
+        if (builtID == null) {
+            builtID = Joiner.on(Const.ID_CONNECTOR).join(fragments);
+        }
+        return builtID;
+    }
+
+    /**
+     * @return a read-only list to avoid unexpected change for metric ID.
+     */
+    public List<Fragment> read() {
+        sealed = true;
+        return Collections.unmodifiableList(fragments);
+    }
+
+    @RequiredArgsConstructor
+    @Getter
+    public static class Fragment {
+        /**
+         * The column name of the value, or the original column names of the mutate value.
+         *
+         * The names could be
+         * 1. Always one column if this is not {@link #mutate} and from a certain persistent column.
+         * 2. Be null if {@link #mutate} is true and no relative column, such as the original value is not in
+         * the persistence.
+         * 3. One or multi-values if the value is built through a symmetrical or asymmetrical encoding algorithm.
+         */
+        private final String[] name;
```

Comment:
@lujiajing1126 @hanahmily The name has been changed to an Array. There are 3 conditions for different cases.

Reply:
Just to confirm the rule,

1. if a model uses mutant fragments, it **MUST NOT** use `@SeriesID` annotation on its fields
2. Otherwise, we must define `@SeriesID` in the model

Reply:
I think @hanahmily has a more complex proposal. @hanahmily Could you share your ideas here?